backend/app/routers/metadata.py

```diff
@@ -70,6 +71,49 @@ async def get_metadata_definition_list(
     return page.dict()
 
 
+@router.put(
+    "/definition/{metadata_definition_id}", response_model=MetadataDefinitionOut
+)
+async def update_metadata_definition(
+    metadata_definition: MetadataDefinitionIn,
+    metadata_definition_id: str,
+    user=Depends(get_current_user),
+):
+    existing_count = await MetadataDefinitionDB.find(
+        {
+            "$and": [
+                {"_id": {"$ne": PydanticObjectId(metadata_definition_id)}},
+                {"name": {"$eq": metadata_definition.name}},
+            ]
+        }
+    ).count()
+    if existing_count > 0:
+        raise HTTPException(
+            status_code=409,
+            detail=f"Metadata definition named {metadata_definition.name} already exists.",
+        )
+    else:
+        if (
+            mdd := await MetadataDefinitionDB.get(
+                PydanticObjectId(metadata_definition_id)
+            )
+        ) is not None:
+            mdd.name = metadata_definition.name
+            mdd.description = metadata_definition.description
+            mdd.required_for_items = metadata_definition.required_for_items
+            mdd.context = metadata_definition.context
+            mdd.context_url = metadata_definition.context_url
+            mdd.fields = metadata_definition.fields
+            mdd.modified = datetime.datetime.utcnow()
```

Comment:
Could this logic be more versatile that we just overwrite everything with metadata_definition without specifically listing each field (except the modified)

Reply:
I have replaced this block by using setattr.

Reply:
Did you actually pushed the setattr part?